src/plugins/visualizations/public/convert_to_lens/schemas.ts

```diff
@@ -31,66 +32,44 @@ const areVisSchemasValid = (visSchemas: Schemas, unsupported: Array<keyof Schema
   return !usedUnsupportedSchemas.length;
 };
 
-export const getColumnsFromVis = <T>(
-  vis: Vis<T>,
-  timefilter: TimefilterContract,
+const createLayer = (
+  visSchemas: Schemas,
+  allMetrics: Array<SchemaConfig<METRIC_TYPES>>,
+  metricsForLayer: Array<SchemaConfig<METRIC_TYPES>>,
+  customBucketsWithMetricIds: Array<{
+    customBucket: IAggConfig;
+    metricIds: string[];
+  }>,
   dataView: DataView,
   {
     splits = [],
     buckets = [],
-    unsupported = [],
   }: {
     splits?: Array<keyof Schemas>;
     buckets?: Array<keyof Schemas>;
-    unsupported?: Array<keyof Schemas>;
   } = {},
-  config?: {
-    dropEmptyRowsInDateHistogram?: boolean;
-  } & (PercentageModeConfig | void)
+  percentageModeConfig: PercentageModeConfig,
+  dropEmptyRowsInDateHistogram?: boolean
 ) => {
-  const { dropEmptyRowsInDateHistogram, ...percentageModeConfig } = config ?? {
-    isPercentageMode: false,
-  };
-  const visSchemas = getVisSchemas(vis, {
-    timefilter,
-    timeRange: timefilter.getAbsoluteTime(),
-  });
-
-  if (!isValidVis(visSchemas) || !areVisSchemasValid(visSchemas, unsupported)) {
-    return null;
-  }
-
-  const customBuckets = getCustomBucketsFromSiblingAggs(visSchemas.metric);
-
-  // doesn't support sibbling pipeline aggs with different bucket aggs
-  if (customBuckets.length > 1) {
+  const metricColumns = metricsForLayer.flatMap((m) =>
+    convertMetricToColumns(m, dataView, allMetrics, percentageModeConfig)
+  );
+  if (metricColumns.includes(null)) {
     return null;
   }
+  const metricColumnsWithoutNull = metricColumns as AggBasedColumn[];
```

Comment:
Why do you need to use `as AggBasedColumn[]` here? Is it because of `meta`?

Reply:
Ok, I remember, that is because of `ts` and its blindness 😃 ) It doesn't respect the previous `includes`.

Reply:
As checking on null above by using 'includes' method don't remove null from type, we should use casting here